rflx/session.py

```diff
@@ -0,0 +1,254 @@
+from typing import Dict, List, Sequence
+
+from rflx.error import Location, RecordFluxError, Severity, Subsystem
+from rflx.expression import (
+    TRUE,
+    Channel,
+    Declaration,
+    Expr,
+    PrivateDeclaration,
+    Renames,
+    Subprogram,
+    VariableDeclaration,
+)
+from rflx.identifier import ID, StrID
+from rflx.model import Base
+from rflx.statement import Statement
+
+
+class StateName(Base):
+    def __init__(self, name: str, location: Location = None):
+        self.name = name
+        self.location = location
+
+    def __hash__(self) -> int:
+        return hash(self.name)
+
+    def __lt__(self, other: object) -> bool:
+        if isinstance(other, StateName):
+            return self.name < other.name
+        return NotImplemented
```

Comment:
Changed in 552e1b0.

Reply:
I just saw that you already removed `StateName` and re-added it before me:

> At least in our current TLS models, state names may have (multiple) dots, which
is incompatible with ID.

I'm not sure how you came to this conclusion: `ID` supports multiple dots. But I tend to disallow dots in state names for now. Dots in identifiers usually have a special meaning and thus could confuse a user. If we decide to support sub-state machines in the future, maybe a dot-notation for states will be needed.

> It sounds preferable, in general, to have a distinct
name space for states anyways.

Maybe you are right. But as the change is already done and the advantages limited, I would suggest keeping it as it is now.

Reply:
> Maybe you are right. But as the change is already done and the advantages limited, I would suggest keeping it as it is now.

Agreed. Thinking about it, I also agree that we should have not dots in state names - this will just confuse users. Let's reconsider that should we add substates.